### PR TITLE
Improve the block storage validator, add 2 fork+sync tests

### PR DESCRIPTION
**Pull request by ljedrz**
This PR improves the block storage validator by not returning in case of any errors, i.e. being more thorough with its analysis, and introduces 2 new tests that simulate a fork with a fairly large, randomized difference of blocks between 2 instances of a ledger and with a few random orphan blocks received before a full sync.

### CODE REVIEW

**Comment by ljedrz** [comment]
The nightly `clippy` error is unrelated.

File: consensus/tests/consensus_sidechain.rs [hunk 1]
```diff
@@ -193,4 +195,117 @@ mod consensus_sidechain {
...
+        // There is no overlap between the 2 instances.
+        let consensus1_locator_hashes = consensus1.ledger.get_block_locator_hashes().unwrap();
+        let latest_shared_hash = consensus2
+            .ledger
+            .get_latest_shared_hash(consensus1_locator_hashes)
+            .unwrap();
+        let shared_height = consensus2.ledger.get_block_number(&latest_shared_hash).unwrap();
+        assert_eq!(shared_height, 0);
+
+        // Consensus 2 imports a few random blocks that consensus 2 has.
```
Comment (niklaslong) [inline consensus/tests/consensus_sidechain.rs:228]:
```suggestion
        // Consensus 1 imports a few random blocks that consensus 2 has.
```

**Review by niklaslong — APPROVED** [review]

File: consensus/tests/consensus_sidechain.rs (hunk 1 above)
Comment (ljedrz) [reply consensus/tests/consensus_sidechain.rs]: good eye :ok:!

**Review by Protryon — APPROVED** [review]

**Comment by ljedrz** [comment]
It seems that the `stable` tests are stuck at fuzzing; we've been getting a lot fewer of them recently, but it can still happen, it seems; I'll restart that job and merge when it finally fails.

**Pull request merged**